src/rez/cli/_entry_points.py

```diff
@@ -0,0 +1,257 @@
+"""
+Entry points.
+"""
+import os.path
+import sys
+
+
+### Utility functions
+
+def get_specifications():
+    """Get entry point specifications
+
+    See:
+    * https://pythonhosted.org/distlib/reference.html#distlib.scripts.ScriptMaker.make_multiple
+    * https://setuptools.readthedocs.io/en/latest/setuptools.html#automatic-script-creation
+
+    Example return value:
+
+        {
+            "rez-env": "rez-env = rez.cli._entry_points.run_rez_env",
+            ...
+        }
+
+    Returns:
+        dict (str, str): The specification string for each script name.
+    """
+    specs = {}
+
+    for attr, obj in sys.modules[__name__].__dict__.iteritems():
+        scriptname = getattr(obj, "__scriptname__", None)
+        if scriptname:
+            spec = "%s = rez.cli._entry_points:%s" % (scriptname, attr)
+            specs[scriptname] = spec
+
+    return specs
+
+
+def scriptname(name):
+    def decorator(fn):
+        setattr(fn, "__scriptname__", name)
+        return fn
+    return decorator
+
+
+def check_production_install():
+    path = os.path.dirname(sys.argv[0])
+    filepath = os.path.join(path, ".rez_production_install")
+
+    if not os.path.exists(filepath):
+        sys.stderr.write(
+            "Pip-based rez installation detected. Please be aware that rez command "
+            "line tools are not guaranteed to function correctly in this case. See "
+            "https://github.com/nerdvegas/rez/wiki/Installation#why-not-pip-for-production "
+            " for futher details.\n"
+        )
+
+
+### Entry points
+
+@scriptname("rez")
+def run_rez():
+    check_production_install()
```

Comment:
I'm wondering if you had a reason to not move that into the `rez.cli._main.run`. I have no problem if we do the check here. The only reason why I was mentioning `rez.cli._main` is because that would allow us to remove some duplication and have the check more centralized. Maybe you see a future where one command would not need the check?